### PR TITLE
chore(migrations): add generated test 0.2 connection records for migration tests

### PR DESCRIPTION
**Pull request by amanji**
This PR is a follow up to #1177 and includes generated test connection records from [this script](https://github.com/petridishdev/aries-framework-javascript-test-data-generator/blob/main/index.ts) along with migration tests and snapshots for AFJ update from 0.2 to 0.3.

### CODE REVIEW

File: packages/core/src/storage/migration/__tests__/__snapshots__/0.2.test.ts.snap [hunk 1]
```diff
@@ -238,6 +238,334 @@ Object {
...
+      "outOfBandId": "b4dc56c1-71ce-46a4-8e8d-ceb8cf81b132",
+      "role": "responder",
+      "state": "response-sent",
+      "theirDid": "did:peer:1zQmQrdQkD5GZZC6JD97Y5ojZMvwNYpTe95FMXF1RHePnrgV",
+      "threadId": "7bbbb76a-cd10-432e-9b61-a322b28dc289",
+    },
+    "type": "ConnectionRecord",
+    "value": Object {
+      "autoAcceptConnection": true,
+      "connectionTypes": Array [
```
Comment (TimoGlastra) [inline packages/core/src/storage/migration/__tests__/__snapshots__/0.2.test.ts.snap:512]: I don't see the mediator connection type added anywhere. The test only has four connections. Maybe we also need a mediator record to see it actually adds them? Also one where we add the mediator connection type in the json beforehand and we see it doesn't add it again?

**Comment by TimoGlastra** [comment]
Hey @amanji, are you planning on finishing this PR? Otherwise I can maybe take a look and get it over the line. I think it would be good to have this merged!

File: packages/core/src/storage/migration/__tests__/__snapshots__/0.2.test.ts.snap (hunk 1 above)
Comment (amanji) [reply packages/core/src/storage/migration/__tests__/__snapshots__/0.2.test.ts.snap:512]: Would this count as a valid mediator connection?

```json
{
  "id": "e171d9d6-54e1-4b28-914f-48d68c668c8b",
  "type": "ConnectionRecord",
  "value": {
    "metadata": {},
    "id": "e171d9d6-54e1-4b28-914f-48d68c668c8b",
    "createdAt": "2023-02-09T20:46:59.943Z",
    "did": "did:peer:1zQmcNtVS8NV6UFRK84ddiGPKhoyyTaZYckh1xZxMF2LAMjb",
    "invitationDid": "did:peer:2.SeyJzIjoiaHR0cHM6Ly9tZWRpYXRvci5kZXYuYW5pbW8uaWQiLCJ0IjoiZGlkLWNvbW11bmljYXRpb24iLCJwcmlvcml0eSI6MCwicmVjaXBpZW50S2V5cyI6WyJkaWQ6a2V5Ono2TWtvSG9RTUphdU5VUE5OV1pQcEw3RGs1SzNtQ0NDMlBpNDJGY3FwR25iampMcSN6Nk1rb0hvUU1KYXVOVVBOTldaUHBMN0RrNUszbUNDQzJQaTQyRmNxcEduYmpqTHEiXSwiciI6W119",
    "theirLabel": "Animo Mediator",
    "state": "completed",
    "role": "requester",
    "autoAcceptConnection": true,
    "threadId": "6e07e8a4-1598-4b06-aca4-1d4459e141cb",
    "protocol": "https://didcomm.org/didexchange/1.0",
    "outOfBandId": "ff0e1e45-fe95-4c15-b0ec-20ee5ce4df90",
    "theirDid": "did:peer:1zQmeDzifN23JzAat4CiPMy9SWbW43X7xKVUyML61EYvk9Gy"
  },
  "tags": {
    "did": "did:peer:1zQmcNtVS8NV6UFRK84ddiGPKhoyyTaZYckh1xZxMF2LAMjb",
    "threadId": "6e07e8a4-1598-4b06-aca4-1d4459e141cb",
    "invitationDid": "did:peer:2.SeyJzIjoiaHR0cHM6Ly9tZWRpYXRvci5kZXYuYW5pbW8uaWQiLCJ0IjoiZGlkLWNvbW11bmljYXRpb24iLCJwcmlvcml0eSI6MCwicmVjaXBpZW50S2V5cyI6WyJkaWQ6a2V5Ono2TWtvSG9RTUphdU5VUE5OV1pQcEw3RGs1SzNtQ0NDMlBpNDJGY3FwR25iampMcSN6Nk1rb0hvUU1KYXVOVVBOTldaUHBMN0RrNUszbUNDQzJQaTQyRmNxcEduYmpqTHEiXSwiciI6W119",
    "outOfBandId": "ff0e1e45-fe95-4c15-b0ec-20ee5ce4df90",
    "role": "requester",
    "theirDid": "did:peer:1zQmeDzifN23JzAat4CiPMy9SWbW43X7xKVUyML61EYvk9Gy",
    "state": "completed",
    "connectionType": ["test-connection-type-1", "test-connection-type-2"]
  }
}
```

Comment (TimoGlastra) [reply packages/core/src/storage/migration/__tests__/__snapshots__/0.2.test.ts.snap:512]: I would say so, yes!

Comment (amanji) [reply packages/core/src/storage/migration/__tests__/__snapshots__/0.2.test.ts.snap:512]: Ok I'll include it and run the tests

**Comment by TimoGlastra** [comment]
Closing this due to inactivity

**Pull request closed without merging**